bundles/framework/myplaces3/CategoryHandler.js

```diff
@@ -66,219 +64,163 @@ Oskari.clazz.define('Oskari.mapframework.bundle.myplaces3.CategoryHandler',
             }
+            return layerId;
+        },
+        getMapLayerId: function (categoryId) {
+            return this.instance.idPrefix + '_' + categoryId;
+        },
+        // Notify other components to update categories data
+        _notifyUpdate: function () {
+            this.sandbox.notifyAll(Oskari.eventBuilder('MyPlaces.MyPlacesChangedEvent')());
         },
         /**
          * @method addLayerToMap
          * Adds the my places map layer to selected if it is not there already
          */
         addLayerToMap: function (categoryId) {
-            var layerId = this._getMapLayerId(categoryId);
+            var layerId = this.getMapLayerId(categoryId);
             var layer = this.sandbox.findMapLayerFromSelectedMapLayers(layerId);
             if (!layer) {
                 var request = Oskari.requestBuilder('AddMapLayerRequest')(layerId, true);
                 this.sandbox.request(this.getName(), request);
             }
         },
+        addLayersToService: function (layers = []) {
+            layers.forEach(layerJson => {
+                this.addLayerToService(layerJson, true);
+            });
+            if (layers.length > 0) {
+                const event = Oskari.eventBuilder('MapLayerEvent')(null, 'add'); // null as id triggers mass update
+                this.sandbox.notifyAll(event);
+            }
+            this._notifyUpdate();
+        },
         /**
          * @method addLayerToService
          * Adds the my places map layer service
          */
-        addLayerToService: function (category) {
+        addLayerToService: function (layerJson, skipEvent) {
             // add maplayer to Oskari
-            var mapLayerService = this.sandbox.getService('Oskari.mapframework.service.MapLayerService');
-            var json = this._getMapLayerJson(category);
-            var myplacesLayer = mapLayerService.createMapLayer(json);
-            mapLayerService.addLayer(myplacesLayer);
-        },
-        updateLayerProperties: function (category) {
-            const mapLayerService = this.sandbox.getService('Oskari.mapframework.service.MapLayerService');
-            const id = this._getMapLayerId(category.getId());
-            const layer = mapLayerService.findMapLayer(id);
-            layer.setName(category.getName());
-            layer.setOptions(category.getOptions());
-        },
-        _getMapLayerId: function (categoryId) {
-            if (!categoryId) {
-                // default to default category id(?)
-                var defCat = this.instance.getService().getDefaultCategory();
-                if (defCat) {
-                    categoryId = defCat.getId();
-                } else {
-                    categoryId = '-99';
-                }
+            const service = this.mapLayerService; // Or getMapLayerService
+            var layer = service.createMapLayer(layerJson);
+            // layer.markAsInternalDownloadSource();
```

Comment:
Is markAsInternalDownloadSource() called somewhere still? This was a workaround so myplaces layers linked in querystring didn't show up on the as the users layers if they were someone elses.

Reply:
Layer doesn't have this method and added comment to remember check if this is needed. Removed comment.